packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ResourceSnapshotsReportModel,
+  ResourceSnapshotsReportOperationModel,
+  ResourceSnapshotsReportResultCode,
+  ResourceSnapshotsReportStatus,
+} from '@coveord/platform-client';
+import {
+  ReportViewerOperationName,
+  ReportViewerResourceReportModel,
+} from './reportViewer/reportViewerDataModels';
+
+export class SnapshotReporter {
+  public constructor(public readonly report: ResourceSnapshotsReportModel) {}
+
+  public hasChangedResources() {
+    const totalUnchanges =
+      this.getOperationTypeTotalCount('resourcesUnchanged');
+    return totalUnchanges !== this.report.resourcesProcessed;
+  }
+
+  public getOperationTypeTotalCount(
+    type: keyof ResourceSnapshotsReportOperationModel
+  ) {
+    const count = Object.values(this.report.resourceOperations).reduce(
+      (count: number, current: ResourceSnapshotsReportOperationModel) =>
+        count + current[type],
+      0
+    );
+
+    return count;
+  }
+
+  public isSuccessReport(): boolean {
+    const {status, resultCode} = this.report;
+    return (
+      status === ResourceSnapshotsReportStatus.Completed &&
+      resultCode === ResourceSnapshotsReportResultCode.Success
+    );
+  }
+
+  public getChangedResources(
+    operationsToDisplay: ReportViewerOperationName[]
+  ): ReportViewerResourceReportModel[] {
+    type resourceEntries = [string, ResourceSnapshotsReportOperationModel];
+    const resourceHasAtLeastOneOperation = ([
+      _,
+      operations,
+    ]: resourceEntries) => {
+      return (
+        operationsToDisplay.reduce(
+          (previous, current) => previous + operations[current],
+          0
+        ) > 0
+      );
+    };
+
+    const convertArrayToObject = ([name, operations]: resourceEntries) => ({
```

Comment:
Semantic, but I think this a tuple, not an array.

But regardless, because you have a named type for your input and output, I would use them for the function name:
`convertResourceEntriesToResourceSnapshotsReportOperationModel`

I agree that it's a mouthful, but I think it describes a bit better what the function does, given the param types and return type.